src/axom/sidre/core/Group.cpp

```diff
@@ -1436,68 +1444,61 @@ void Group::load(const std::string& path,
     importConduitTree(n, preserve_contents);
     if (n.has_path("sidre_group_name"))
     {
-      new_name = n["sidre_group_name"].as_string();
+      name_from_file = n["sidre_group_name"].as_string();
     }
   }
   else
   {
     SLIC_ERROR("Invalid protocol " << protocol << " for file load.");
   }
+}
 
-  if (!new_name.empty())
+/*
+ *************************************************************************
+ *
+ * Load Group (including Views and child Groups) from a file
+ *
+ *************************************************************************
+ */
+Group* Group::createAndLoad(std::string & group_name,
+                            const std::string& path,
+                            const std::string& protocol,
+                            bool & load_success)
+{
+  load_success = false;
+  Group * child = createGroup(group_name);
+  if (child != nullptr)
   {
-    renameOrWarn(new_name);
+    // In a forthcoming PR, load() will return a bool for success/failure
+    load_success = true;
+    child->load(path, protocol, false, group_name);
   }
+  
+  return child;
 }
 
+
 /*
  *************************************************************************
  *
- * Rename this group unless the new name already is held by the parent.
+ * Find a valid (unique) name for a new group
  *
  *************************************************************************
  */
-void Group::renameOrWarn(const std::string& new_name)
+std::string Group::getUniqueGroupName(const std::string & basename) const
 {
-  if (new_name != m_name)
+  int counter = 0;
+  std::string name = basename;
+  while (hasGroup(name))
```

Comment:
Is getUniqueGroupName needed anymore?  Looks like all usage was taken out in the latest commits.  If it is needed, we should check here on both hasGroup() and hasView() as either one can create a naming conflict.

Reply:
I don't think `getUniqueGroupName()` is needed, since I took the last call out.  I just forgot to take out the function itself.

Reply:
done

Reply:
I think we might still need this as a public utility function -- possibly as a free function, since you still needs to call ``createAndLoad()`` with a valid name (i.e. not one of the child group or view names).  Without this utility function, users will need to write their own.